docs/walkthrough/using-callbacks.md

```diff
@@ -0,0 +1,128 @@
+
+## BestModelCheckpoint
+
+This callback evaluates the performance of the model at the end of each epoch, and keeps a record of the best perfoming model across all epochs. Once fitting is finished the best performing model is saved instead of the most recent model. The definition of best is based on two parameters:
+
+- `monitor`: The metric that is used to compare models to each other. By default this value is 'val_loss', the loss function calculated using the evaluation data, however the loss calculated on the training data can be used instead with 'train_loss'; any metric that is recorded by the evaluation callback can also be used.
+- `mode`: Whether the monitored metric should be maximised ('max') or minimised ('min'). By default the mode is set to auto, meaning that it will automatically chose the correct mode depending on the chosen metric: 'min' if the metric is loss and 'max' if the metric is one recorded by the evaluation callback.
+
+The console output below shows how the evaluation loss of the model is monitored between each epoch, and how the best performing model is tracked. Since the final model has a higher loss than the previously recorded best model, the best model will be saved instead of the latest one.
+
+```bash
+           INFO     Finetuning ...                                                   
+                    __main__.py:173
+[11:50:33] INFO     Model improved from inf to 2.756!                                
+       best_model_checkpoint.py:112
+[11:50:52] INFO     Model improved from 2.756 to 2.711!                              
+       best_model_checkpoint.py:112
+[11:51:10] INFO     Model did not improve                                            
+       best_model_checkpoint.py:120
+[11:51:28] INFO     Model did not improve                                            
+       best_model_checkpoint.py:120
+  Training [4/4] ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54/54 0:00:00 0:00:15 • loss: 0.496 • val_loss: 2.797
+           INFO     Done ✨                                                          
+                    __main__.py:194
+           DEBUG    Finetuning took 0 days, 0 hours 1 minutes and 16 seconds         
+                    __main__.py:196
+           INFO     Building the artifact ...                                        
+                    __main__.py:207
+           INFO     Pushing artifact to Hubble ...                                   
+                    __main__.py:231
+```
+
+This callback is triggered at the end of both training and evaluation batches, to record the losses of the two data sets, and is triggered a third time at the end of each epoch to evaluate the performance of the current model using the monitored metric and then record this model if it performs better than the best model so far.

```

Comment:
I don't know if it is relevant to the user to know what exactly each event causes in the callback. I think the information which is important for the user is already mentioned above. 

If we want to keep this sentence, I would suggest to split it into multiple shorter sentences. Moreover, it should be a little bit more precise, i.e., a third time is technically not correct since it is called before after each batches, and, at the end of the fit process it is also called another time to load the best model.

Reply:
I agree and have removed this part now